app/amity.py

```diff
@@ -0,0 +1,327 @@
+        # Check if there are rooms
+        if len(self.list_of_rooms):
+            # Make a list of available rooms depending on person role
+            available_rooms = []
+            for room in self.list_of_rooms:
+                if (room.room_type == "Office" and not room.full()) \
+                        or (room.room_type == "Living Space" and not room.full()):
+                    available_rooms.append(room)
+
+            if person.role == "Staff":
+                available_rooms = [room for room in available_rooms if
+                                   room.room_type == "Office" and not room.full()]
+            if available_rooms:
+                room = random.choice(available_rooms)
+            else:
+                pass
+            while not allocated:
+                # Allocate person a room randomly
+                if room.room_type == "Living Space" and not room.full() \
+                        and person.role == "Fellow" and wants_accomodation == 'Y':
+                    room.occupants.append(person)
+                    allocated = True
+                    break
+                elif room.room_type == "Living Space" and not room.full() \
+                        and person.role == "Fellow":
+                    room.occupants.append(person)
+                    allocated = True
+                    break
+                elif room.room_type == "Office" and not room.full():
+                    room.occupants.append(person)
+                    allocated = True
+                    break
+                else:
+                    break
+        else:
+            cprint("There are no rooms to allocate", 'yellow')
+
+        if not allocated:
+            self.unallocated_members.append(person)
+            cprint("Sorry {}. We couldn't find a suitable room for you"
+                   .format(person.person_name), 'yellow')
+            message = ("{} was added to the waiting list".format(
+                person.person_name))
+            cprint(message, 'green')
+        else:
+            person.current_room = room.room_name
+            self.allocated_members.append(person)
+            message = ("{} successfully allocated {}".format(
+                person.person_name, room.room_name))
+            cprint(message, 'green')
+
+        return message
+
+    def reallocate_person(self, person_name, room_name):
+        """Transfers person to a different room."""
+
+        room = [room for room in self.list_of_rooms
+                if room_name.upper() == room.room_name.upper()]
+        person = [person for person in self.allocated_members
+                  if person_name.upper() == person.person_name.upper()]
+        if not person:
+            cprint("{} has no room in Amity".format(person_name), 'yellow')
+            return "Person not found"
+        elif not room:
+            cprint("{} is not a room in Amity".format(room_name), 'yellow')
+            return "Room not found"
```

Comment:
for the block of code 122-123 and 125-126 choose to have one simple message that is returned or printed on the screen. What is there currently is confusing.